mslice/plotting/plot_window/cut_plot.py

```diff
@@ -321,15 +325,28 @@ def set_is_icut(self, is_icut):
     def is_icut(self):
         return self._is_icut
 
-    def update_bragg_peaks(self):
+    def _get_overplot_datum(self):
+        return np.nanmedian([line.get_ydata() for line in self._canvas.figure.gca().get_lines()
+                             if not self._cut_plotter_presenter.is_overplot(line)])
+
+    def update_bragg_peaks(self, refresh=None):
+        refresh = False if refresh is None else True
         if self.plot_window.action_aluminium.isChecked():
-            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Aluminium', False, cif=None)
+            refresh and self._cut_plotter_presenter.hide_overplot_line(None, 'Aluminium')
+            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Aluminium', False, None, self.y_log,
+                                                          self._get_overplot_datum())
         if self.plot_window.action_copper.isChecked():
-            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Copper', False, cif=None)
+            refresh and self._cut_plotter_presenter.hide_overplot_line(None, 'Copper')
+            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Copper', False, None, self.y_log,
+                                                          self._get_overplot_datum())
         if self.plot_window.action_niobium.isChecked():
-            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Niobium', False, cif=None)
+            refresh and self._cut_plotter_presenter.hide_overplot_line(None, 'Niobium')
+            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Niobium', False, None, self.y_log,
+                                                          self._get_overplot_datum())
         if self.plot_window.action_tantalum.isChecked():
-            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Tantalum', False, cif=None)
+            refresh and self._cut_plotter_presenter.hide_overplot_line(None, 'Tantalum')
+            self._cut_plotter_presenter.add_overplot_line(self.ws_name, 'Tantalum', False, None, self.y_log,
+                                                          self._get_overplot_datum())
```

Comment:
`self._get_overplot_datum()` looks like it could be expensive if its called for each of the bragg peaks. Do you think doing this calculation once, above the if statements, would be worth doing (even when none of the bragg peaks are checked)?

Reply:
Sounds good, lets just leave it as is then 👍

Reply:
We could cache the median, and recalculate only upon line removal/addition. I initially thought this would be too much of a pain, but thinking of this makes we realise we probably need to refresh the bragg peaks at this point anyway as the datum will change.